src/concrete/ml/quantization/quantized_module.py

```diff
@@ -501,10 +514,23 @@ def _fhe_forward(self, *q_x: numpy.ndarray, simulate: bool = True) -> numpy.ndar
             # Execute the forward pass in FHE or with simulation
             q_result = predict_method(*q_input)
 
-            results_cnp_circuit_list.append(q_result)
-
-        results_cnp_circuit = numpy.concatenate(results_cnp_circuit_list, axis=0)
-
+            if isinstance(q_result, tuple):
```

Comment:
similar to above, I think it would be simpler if you use the `to_tuple` function here , which would make the code look like : 
```
q_result = to_tuple(q_result)

results_cnp_circuit_list = [[] for _ in q_result]

for elt_index, elt in enumerate(q_result):
    results_cnp_circuit_list[elt_index].append(elt)
```

Reply:
yeah makes sense, much better indeed!